sktime_dl/deeplearning/cnn/_classifier.py

```diff
@@ -15,67 +16,58 @@ class CNNClassifier(BaseDeepClassifier, CNNNetwork):
+    :param batch_size: int, the number of samples per gradient update.
+    :param kernel_size: int, specifying the length of the 1D convolution
+     window
+    :param avg_pool_size: int, size of the average pooling windows
+    :param nb_conv_layers: int, the number of convolutional plus average
+     pooling layers
+    :param filter_sizes: int, array of shape = (nb_conv_layers)
+    :param callbacks: list of tf.keras.callbacks.Callback objects
+    :param random_seed: int, seed to any needed random actions
+    :param verbose: boolean, whether to output extra information
+    :param model_name: string, the name of this model for printing and
+    file writing purposes
+    :param model_save_directory: string, if not None; location to save
+    the trained keras model in hdf5 format
     """
 
     def __init__(
-        self,
-        nb_epochs=2000,
-        batch_size=16,
-        kernel_size=7,
-        avg_pool_size=3,
-        nb_conv_layers=2,
-        filter_sizes=[6, 12],
-        callbacks=None,
-        random_seed=0,
-        verbose=False,
-        model_name="cnn",
-        model_save_directory=None,
-    ):
-        super().__init__(
-            model_name=model_name, model_save_directory=model_save_directory
-        )
-        CNNNetwork.__init__(
             self,
-            kernel_size=kernel_size,
-            avg_pool_size=avg_pool_size,
-            nb_conv_layers=nb_conv_layers,
-            filter_sizes=filter_sizes,
-            random_seed=random_seed,
-        )
-        """
-        :param nb_epochs: int, the number of epochs to train the model
-        :param batch_size: int, the number of samples per gradient update.
-        :param kernel_size: int, specifying the length of the 1D convolution
-         window
-        :param avg_pool_size: int, size of the average pooling windows
-        :param nb_conv_layers: int, the number of convolutional plus average
-         pooling layers
-        :param filter_sizes: int, array of shape = (nb_conv_layers)
-        :param callbacks: list of tf.keras.callbacks.Callback objects
-        :param random_seed: int, seed to any needed random actions
-        :param verbose: boolean, whether to output extra information
-        :param model_name: string, the name of this model for printing and
-        file writing purposes
-        :param model_save_directory: string, if not None; location to save
-        the trained keras model in hdf5 format
-        """
+            nb_epochs=2000,
+            batch_size=16,
+            kernel_size=7,
+            avg_pool_size=3,
+            nb_conv_layers=2,
+            filter_sizes=[6, 12],
+            callbacks=None,
+            random_seed=0,
+            verbose=False,
+            model_name="cnn",
+            model_save_directory=None,
+    ):
+        super(CNNClassifier, self).__init__(
+            model_save_directory=model_save_directory,
+            model_name=model_name)
+        self.filter_sizes = filter_sizes
```

Comment:
Yes, I chatted with one of the core devs on this issue on Gitter (can't link the post, but you can scroll up to Apr 27 09:34 to find my question). Including other attributes shouldn't break anything, as long as there is no interaction/logic between attributes/input arguments, all of that should happen in fit.

Reply:
It's because we initialise other attributes like `self.classes` and so on in the `BaseDeepClassifier`, I'm still not entirely happy with how we solve the multiple inheritance, because there's still a lot of duplication in the constructors, but that's perhaps something for another issue.

Reply:
Yes, ok.